### ISSUE TITLE
[java] GuardLogStatement false positive on return statements and Math.log

### ISSUE THREAD

**Issue by vanniktech** [pmd/pmd#869]
**Affects PMD Version:** 

6.0.1 (did not happen with 6.0.0)

**Rule:**

GuardLogStatement

**Description:**

The code below gets flagged although it should not since I'm not using any exception.

**Code Sample demonstrating the issue:**

```java
return Single.error(new IllegalStateException("No permission was pre-configured for " + permission));
```

You can also check out this PR - https://github.com/vanniktech/RxPermission/pull/21 - which fails

**Report**

```xml
<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd"
    version="6.0.1" timestamp="2018-01-21T19:15:40.307">
<file name="/Users/nik/tmp/RxPermission/library/src/main/java/com/vanniktech/rxpermission/MockRxPermission.java">
<violation beginline="32" endline="32" begincolumn="12" endcolumn="104" rule="GuardLogStatement" ruleset="Best Practices" package="com.vanniktech.rxpermission" class="MockRxPermission" method="request" externalInfoUrl="https://pmd.github.io/pmd-6.0.1/pmd_rules_java_bestpractices.html#guardlogstatement" priority="2">
There is log block not surrounded by if
</violation>
</file>
</pmd>
```

**Running PMD through:** *[Gradle]*

**Comment by jsotuyod** [comment]
Interesting... seems like #806 fixed #783 but introduced this...

**Comment by ben-manes** [comment]
I got a false positive with `Math.log`...

```java
this.depth = (int) Math.ceil(-Math.log(1 - confidence) / Math.log(2));
```
